File: solutions/elixir/isogram/1/lib/isogram.ex
defmodule Isogram do
  @doc """
  Determines if a word or sentence is an isogram
  """
  @spec isogram?(String.t()) :: boolean
  def isogram?(sentence) do
    sentence
    |> String.downcase()
    |> String.graphemes()
    |> Enum.filter(& &1 not in ["-", " "])
    |> then(& Enum.count(&1) === MapSet.new(&1) |> Enum.count())
  end
end

File: solutions/ruby/isogram/1/isogram.rb
# frozen_string_literal: true

# Determine if a word or phrase is an isogram.
module Isogram
  def self.isogram?(word)
    chars = word.downcase.chars.filter { |c| !['-', ' '].include?(c) }
    chars.uniq.eql?(chars)
  end
end
